xgi/classes/dihypergraph.py

```diff
@@ -0,0 +1,823 @@
+            word = "edges"
+        if stat is None:
+            word = None
+            raise AttributeError(
+                f"{attr} is not a method of Hypergraph or a recognizedDiNodeStat or DiEdgeStat"
+            )
+
+        def func(node=None, *args, **kwargs):
+            val = stat(*args, **kwargs).asdict()
+            return val if node is None else val[node]
+
+        func.__doc__ = f"""Equivalent to H.{word}.{attr}.asdict(). For accepted *args and
+        **kwargs, see documentation of H.{word}.{attr}."""
+
+        return func
+
+    @property
+    def num_nodes(self):
+        """The number of nodes in the hypergraph.
+
+        Returns
+        -------
+        int
+            The number of nodes in the hypergraph.
+
+        See Also
+        --------
+        num_edges : returns the number of edges in the hypergraph
+
+        Examples
+        --------
+        >>> import xgi
+        >>> hyperedge_list = [([1, 2], [2, 3, 4])]
+        >>> H = xgi.DiHypergraph(hyperedge_list)
+        >>> H.num_nodes
+        4
+
+        """
+        return len(self._node_in)
+
+    @property
+    def num_edges(self):
+        """The number of directed edges in the hypergraph.
+
+        Returns
+        -------
+        int
+            The number of directed edges in the hypergraph.
+
+        See Also
+        --------
+        num_nodes : returns the number of nodes in the hypergraph
+
+        Examples
+        --------
+        >>> import xgi
+        >>> hyperedge_list = [([1, 2], [2, 3, 4])]
+        >>> H = xgi.DiHypergraph(hyperedge_list)
+        >>> H.num_edges
+        1
+        """
+        return len(self._edge_in)
+
+    @property
+    def nodes(self):
+        """A :class:`DiNodeView` of this network."""
+        return self._nodeview
+
+    @property
+    def edges(self):
+        """An :class:`DiEdgeView` of this network."""
+        return self._edgeview
+
+    def add_node(self, node, **attr):
+        """Add one node with optional attributes.
+
+        Parameters
+        ----------
+        node : node
+            A node can be any hashable Python object except None.
+        attr : keyword arguments, optional
+            Set or change node attributes using key=value.
+
+        See Also
+        --------
+        add_nodes_from
+
+        Notes
+        -----
+        If node is already in the hypergraph, its attributes are still updated.
+
+        """
+        if node not in self._node_in:
+            self._node_in[node] = set()
+            self._node_out[node] = set()
+            self._node_attr[node] = self._node_attr_dict_factory()
+        self._node_attr[node].update(attr)
+
+    def add_nodes_from(self, nodes_for_adding, **attr):
+        """Add multiple nodes with optional attributes.
+
+        Parameters
+        ----------
+        nodes_for_adding : iterable
+            An iterable of nodes (list, dict, set, etc.).
+            OR
+            An iterable of (node, attribute dict) tuples.
```

Comment:
we don't have this option for `Hypergraph` right? Should we add it for consistency?

Reply:
I totally agree! If I understand you correctly, this is raised in Issue [#342](https://github.com/xgi-org/xgi/issues/342) and I can address that once this PR is merged.